pallene/to_ir.lua

```diff
@@ -209,60 +209,99 @@ function ToIR:convert_stat(cmds, stat)
         local decls = stat.decls
         local exps = stat.exps
 
+        local is_ipairs = false
+        local iterfn = exps[1]
+        if iterfn._tag == "ast.Exp.CallFunc" and iterfn.exp._tag == "ast.Exp.Var" then
+            local iterfn_var = iterfn.exp.var._name
+            is_ipairs = iterfn_var._tag == "checker.Name.Builtin" and iterfn_var.name == "ipairs"
+        end
```

Comment:
The variable names here might be a bit misleading. The iterfn refers to a function *call*, not to a function. And the iterfn_var refers to a Checker.Name, not to a Var. What do you think about just getting rid of those variables?

```
local e1 = exps[1]
local is_ipairs = (
    e1._tag == "ast.Exp.CallFunc" and
    e1.exp._tag == "ast.Exp.Var" and
    e1.exp.var._name._tag == "checker.Name.Builtin" and
    e1.exp.var._name.name == "ipairs")
```

Reply:
Yeah, This looks good. I initially was under the impression that using those names would make it easier to read but I think this looks better and reads more naturally.